File: labs/lesson-002/model_details.rb
require 'aws-sdk-bedrock'

begin
  client = Aws::Bedrock::Client.new(region: 'us-east-1')

  model_id = ARGV[0] || 'anthropic.claude-3-5-sonnet-20241022-v2:0'
  puts "Fetching details for model: #{model_id}..."

  response = client.get_foundation_model(model_identifier: model_id)
  details = response.model_details

  puts "\nModel Details Summary:"
  puts "=" * 50
  puts "Name:         #{details.model_name}"
  puts "ID:           #{details.model_id}"
  puts "Provider:     #{details.provider_name}"
  puts "Input Types:  #{details.input_modalities.join(', ')}"
  puts "Output Types: #{details.output_modalities.join(', ')}"
  puts "Customizable: #{details.customizations_supported.join(', ')}"
  puts "Streaming:    #{details.response_streaming_supported ? 'Yes' : 'No'}"

rescue Aws::Bedrock::Errors::ValidationException
  puts "Error: Model ID '#{model_id}' is not recognized or not available in us-east-1."
rescue Aws::Bedrock::Errors::ServiceError => e
  puts "AWS Bedrock Error: #{e.message}"
rescue StandardError => e
  puts "Error: #{e.message}"
end

File: labs/lesson-002/model_details.js
const { BedrockClient, GetFoundationModelCommand } = require("@aws-sdk/client-bedrock");

async function checkModel() {
  const modelId = process.argv[2] || "anthropic.claude-3-5-sonnet-20241022-v2:0";
  console.log(`Fetching details for model: ${modelId}...`);

  const client = new BedrockClient({ region: "us-east-1" });

  try {
    const command = new GetFoundationModelCommand({ modelIdentifier: modelId });
    const response = await client.send(command);

    const details = response.modelDetails;
    console.log("\nModel Details Summary:");
    console.log("=".repeat(50));
    console.log(`Name:         ${details.modelName}`);
    console.log(`ID:           ${details.modelId}`);
    console.log(`Provider:     ${details.providerName}`);
    console.log(`Input Types:  ${details.inputModalities.join(", ")}`);
    console.log(`Output Types: ${details.outputModalities.join(", ")}`);
    console.log(`Streaming:    ${details.responseStreamingSupported ? "Yes" : "No"}`);

  } catch (error) {
    if (error.name === "ValidationException") {
      console.log(`Error: Model ID "${modelId}" is not recognized or not available in this region.`);
    } else {
      console.error("AWS Error:", error.message || error);
    }
  }
}

checkModel();
